framework_tvb/tvb/adapters/analyzers/fourier_adapter.py

```diff
@@ -214,15 +243,33 @@ def launch(self, view_model):
         fft_index.ndim = len(spectra_file.array_data.shape)
         input_time_series_h5.close()
 
-        fft_index.windowing_function = self.algorithm.window_function
-        fft_index.segment_length = self.algorithm.segment_length
-        fft_index.detrend = self.algorithm.detrend
+        fft_index.windowing_function = view_model.window_function
+        fft_index.segment_length = partial_result.segment_length
+        fft_index.detrend = view_model.detrend
         fft_index.frequency_step = partial_result.freq_step
         fft_index.max_frequency = partial_result.max_freq
 
-        spectra_file.segment_length.store(self.algorithm.segment_length)
-        spectra_file.windowing_function.store(str(self.algorithm.window_function))
+        spectra_file.segment_length.store(view_model.segment_length)
```

Comment:
Can you check whether we can also do `spectra_file.store(partial_result, scalars_only=True)` to take care of the scalars storage in one line? Then we would need to leave only the `spectra_file.write_data_slice(partial_result)` to store each computed slice in a dataset.